src/backend/distributed/utils/metadata_cache.c

```diff
@@ -1381,9 +1358,10 @@ HasOverlappingShardInterval(ShardInterval **shardIntervalArray,
 		Assert(lastShardInterval->minValueExists && lastShardInterval->maxValueExists);
 		Assert(curShardInterval->minValueExists && curShardInterval->maxValueExists);
 
-		comparisonDatum = CompareCall2(shardIntervalSortCompareFunction,
-									   lastShardInterval->maxValue,
-									   curShardInterval->minValue);
+		comparisonDatum = FunctionCall2Coll(shardIntervalSortCompareFunction,
+											DEFAULT_COLLATION_OID,
```

Comment:
why are we using the `DEFAULT_COLLATION_OID` here and not the distribution column's collation?